components/collector/src/source_collectors/quality_time/missing_metrics.py

```diff
@@ -37,15 +37,21 @@ def _include_entity(self, entity: Entity) -> bool:
         """Return whether to include the entity in the measurement."""
         source_types_to_include = set(self._parameter("source_types_to_include"))
         metric_source_types = set(self.data_model["metrics"][entity["metric_type"]]["sources"])
-        if source_types_to_include and not (source_types_to_include & metric_source_types):
+        if not (source_types_to_include & metric_source_types):
             return False
-        subjects_to_ignore = self._parameter("subjects_to_ignore")
-        if not subjects_to_ignore:
-            return True
-        return not match_string_or_regular_expression(
-            entity["subject"],
-            subjects_to_ignore,
-        ) and not match_string_or_regular_expression(entity["subject_uuid"], subjects_to_ignore)
+
+        if subjects_to_include := self._parameter("subjects_to_include"):
+            subject_name_matches = match_string_or_regular_expression(entity["subject"], subjects_to_include)
+            subject_uuid_matches = match_string_or_regular_expression(entity["subject_uuid"], subjects_to_include)
+            if not (subject_name_matches or subject_uuid_matches):
+                return False
+
+        if subjects_to_ignore := self._parameter("subjects_to_ignore"):
```

Comment:
Is it possible for the partial name of any subject to be included within another subject? I.e. that it would be included and then also ignored due to partial regex match?